plugins/config/config.go

```diff
@@ -2,29 +2,156 @@ package config
+	if err != nil {
+		return defaultValue
+	}
+	return env.GetDefault(key, defaultValue)
+}
+
 // GetWithDefault returns the value set for a given key, returning defaultValue if none found
-func GetWithDefault(appName string, key string, defaultValue string) (value string) {
-	value = defaultValue
+// func GetWithDefault(appName string, key string, defaultValue string) (value string) {
+// 	value = defaultValue
+
+// 	envFile := strings.Join([]string{common.MustGetEnv("DOKKU_ROOT"), appName, "ENV"}, "/")
+// 	lines, err := common.FileToSlice(envFile)
+// 	if err != nil {
+// 		return
+// 	}
+// 	prefix := fmt.Sprintf("export %v=", key)
+// 	for _, line := range lines {
+// 		if !strings.HasPrefix(line, prefix) {
+// 			continue
+// 		}
+// 		value = strings.TrimPrefix(line, prefix)
+// 		if strings.HasPrefix(value, "'") && strings.HasSuffix(value, "'") {
+// 			value = strings.TrimPrefix(strings.TrimSuffix(value, "'"), "'")
+// 		}
+// 	}
+// 	return
+// }
 
-	envFile := strings.Join([]string{common.MustGetEnv("DOKKU_ROOT"), appName, "ENV"}, "/")
-	lines, err := common.FileToSlice(envFile)
+//HasKey determines if the config given by appName has a value for the given key
+func HasKey(appName string, key string) bool {
+	env, err := loadConfig(appName)
 	if err != nil {
-		return
+		return false
+	}
+	_, ok := env.Get(key)
+	return ok
+}
+
+//SetMany variables in the environment. If appName is empty the global config is used. If restart is true the app is restarted.
+func SetMany(appName string, entries map[string]string, restart bool) {
+	global := appName == ""
+	env := GetConfig(appName, false)
+
+	keys := make([]string, 0, len(entries))
+
+	for k, v := range entries {
+		env.Set(k, v)
+		keys = append(keys, k)
+	}
+
+	if len(entries) != 0 {
+		common.LogInfo1("Setting config vars")
+		fmt.Println(PrettyPrintEnvEntries("       ", entries))
+		env.Write()
+		args := append([]string{appName, "set"}, keys...)
+		common.PlugnTrigger("post-config-update", args...)
+	}
+
+	if !global && restart && env.GetBoolDefault("DOKKU_APP_RESTORE", true) {
+		Restart(appName)
+	}
+}
+
+//UnsetMany a value in a config. If appName is empty the global config is used. If restart is true the app is restarted.
+func UnsetMany(appName string, keys []string, restart bool) {
+	global := appName == ""
+	env := GetConfig(appName, false)
+	var changed = false
+	for _, k := range keys {
+		common.LogInfo1(fmt.Sprintf("Unsetting %s", k))
+		env.Unset(k)
+		changed = true
+	}
+
+	if changed {
+		env.Write()
+		args := append([]string{appName, "unset"}, keys...)
+		common.PlugnTrigger("post-config-update", args...)
 	}
-	prefix := fmt.Sprintf("export %v=", key)
-	for _, line := range lines {
-		if !strings.HasPrefix(line, prefix) {
-			continue
+
+	if !global && restart && env.GetBoolDefault("DOKKU_APP_RESTORE", true) {
+		Restart(appName)
+	}
+}
+
+//PrettyPrintEnvEntries in columns
+func PrettyPrintEnvEntries(prefix string, entries map[string]string) string {
```

Comment:
does this need to be exported?

Reply:
Bare `dokku config` uses it to print out the environment right now.